### DISCUSSION TITLE
SSE not working in post method

### DISCUSSION THREAD

**Discussion by jamesljl** [kartikk221/hyper-express#292] [General]
`
export const index_router = new Router();

index_router.get("/sse", (request, response) => {
    let i = 0;
    if (response.sse) {
        console.log("client got connected");
        response.once("close", () => {
            console.log("client got disconnected");
        });

        response.sse.open();
        response.sse.send("sse connection open");
        const h = setInterval(() => {
            i++;
            if (i >= 5) {
                response.sse.close();
                return
            }
            response.sse.send("event sent");
        }, 1000);
    } else {
        response.send("SSE not supported");
    }
});

`
above code just works fine. But once change it to index_router.post("/sse", (request, response) => {}, error raised up says : SSE not supported.
Is it a bug or something I just missed?

**Comment by kartikk221** [comment]
Currently HyperExpress only enables SSE on GET requests as it was built for Browser EventSource compatibility which only supports GET requests.

I am assuming you are trying to do some sort of an unconventional POST request SSE which acts as a stream?

**Reply by jamesljl** [reply]
thx for ur reply. I am just trying to send stream data to browser in post method and failed many times using response.write().

**Reply by jamesljl** [reply]
my fault, I just get things done. thx again.
